src/plugins/exmon/exmon.c

```diff
@@ -0,0 +1,203 @@
+ * alternative license can be requested from the author to integrate       *
+ * DRAKVUF into proprietary applications and appliances.  Please email     *
+ * tamas.k.lengyel@gmail.com for further information.                      *
+ *                                                                         *
+ * If you have received a written license agreement or contract for        *
+ * Covered Software stating terms other than these, you may choose to use  *
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <glib.h>
+#include <libvmi/libvmi.h>
+#include "private.h"
+#include "../plugins.h"
+
+static GSList *traps;
+static output_format_t format;
+
+static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+    KTRAP_FRAME trap_frame;
+    reg_t exception_record, ptrap_frame, exception_code;
+
+    page_mode_t pm = vmi_get_page_mode(vmi);
```

Comment:
If you don't expect the pagemode to change at runtime (you are passed the OS boot) you could just read the pagemode once in the plugin init and store in the a global.